exporter/datadogexporter/config.go

```diff
@@ -0,0 +1,170 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadogexporter
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/confignet"
+)
+
+var (
+	errUnsetAPIKey = errors.New("the Datadog API key is unset")
+)
+
+// APIConfig defines the API configuration options
+type APIConfig struct {
+	// Key is the Datadog API key to associate your Agent's data with your organization.
+	// Create a new API key here: https://app.datadoghq.com/account/settings
+	Key string `mapstructure:"key"`
+
+	// Site is the site of the Datadog intake to send data to.
+	// The default value is "datadoghq.com".
+	Site string `mapstructure:"site"`
+}
+
+// GetCensoredKey returns the API key censored for logging purposes
+func (api *APIConfig) GetCensoredKey() string {
+	if len(api.Key) <= 5 {
+		return api.Key
+	}
+	return strings.Repeat("*", len(api.Key)-5) + api.Key[len(api.Key)-5:]
+}
+
+// MetricsConfig defines the metrics exporter specific configuration options
+type MetricsConfig struct {
+	// Namespace is the namespace under which the metrics are sent
+	// By default metrics are not namespaced
+	Namespace string `mapstructure:"namespace"`
+
+	// Buckets states whether to report buckets from distribution metrics
+	Buckets bool `mapstructure:"report_buckets"`
+
+	// TCPAddr.Endpoint is the host of the Datadog intake server to send metrics to.
+	// If unset, the value is obtained from the Site.
+	confignet.TCPAddr `mapstructure:",squash"`
+}
+
+// TagsConfig defines the tag-related configuration
+// It is embedded in the configuration
+type TagsConfig struct {
+	// Hostname is the host name for unified service tagging.
+	// If unset, it is determined automatically.
+	// See https://docs.datadoghq.com/agent/faq/how-datadog-agent-determines-the-hostname
+	// for more details.
+	Hostname string `mapstructure:"hostname"`
+
+	// Env is the environment for unified service tagging.
+	// It can also be set through the `DD_ENV` environment variable.
+	Env string `mapstructure:"env"`
+
+	// Service is the service for unified service tagging.
+	// It can also be set through the `DD_SERVICE` environment variable.
+	Service string `mapstructure:"service"`
+
+	// Version is the version for unified service tagging.
+	// It can also be set through the `DD_VERSION` version variable.
+	Version string `mapstructure:"version"`
+
+	// Tags is the list of default tags to add to every metric or trace.
+	Tags []string `mapstructure:"tags"`
+}
+
+// UpdateWithEnv gets the unified service tagging information
+// from the environment variables.
+func (t *TagsConfig) UpdateWithEnv() {
```

Comment:
We already support loading env variables, and we would expect you to use the same mechanism. You can simply make the default `Hostname = "${DD_HOST}"`.

So I would suggest this custom mechanism to be removed

Reply:
I changed it to use the existing mechanism, thanks for the comment!